include/zephyr/drivers/spi.h

```diff
@@ -1306,6 +1307,35 @@ static inline int spi_release_dt(const struct spi_dt_spec *spec)
 	return spi_release(spec->bus, &spec->config);
 }
 
+/**
+ * @brief Checks that all buffers in `bufs` are in nocache memory region. This
+ * should be used in platforms where DMA cannot access cache memory, and so any
+ * buffer used for DMA operations must be in a nocache region.
```

Comment:
I agree; that would require a new SPI private header file common to all SPI drivers, as this is "made public" here because it was deemed of general use in changes requested above.

Reply:
Maybe this one function and one macro would justify creating a `drivers/spi/spi.h` file to move private parts there. Personally, I would create it even if there was only one function to clearly separate "public" from "private". But I believe that's a personal taste :)